### PR TITLE
V6 component tooltip docs

### PR DESCRIPTION
**Pull request by dialpad-drew**
## Description
- Backmerges with the `version6` branch so we can use some of the new shortcodes merged in earlier today
- Re-organizes tooltip's written documentation based on @ryandialpad's CSS refactor
- Creates `tooltips.json` for classes and accessibility table content; this is in-line with other components' content organization
- Re-styles documentation page

### Other notes

1. Something I'm unsure about in here is the tooltip going _over_ the site-nav. This is expected behavior from using the defined `z-index` values in our utilities and makes sense for tooltips that are being used on the doc site (i.e. the ones in the header). However, the tooltips being demonstrated on their documentation page _also_ go "above" the header, which appears like a UI defect to me. Does anyone else feel this way? I'm wondering if this is another example where the doc site simply using the CSS meant for the rest of Dialpad doesn't necessarily work.
2. I'm also wondering if we need a `<table>` shortcode, since just about every utility or component will use some version of the same table. It seems incredibly redundant to repeat that mark-up on a few dozen pages.
3. The accessibility content included here is mostly FPO. Though I believe this _could_ make tooltips more accessible, it's meant to be updated when the Vue component has been finished. Called out in #203

### CODE REVIEW

**Comment by dialpad-drew** [comment]
@ryandialpad what I'd particularly like your blessing on is whether I accurately listed and described the necessary CSS classes in the "Classes" table. Obviously any other thoughts or feedback are welcome if you have 'em.

File: docs/_data/tooltips.json [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+    },
+    {
+      "class": "d-tooltip--hide",
+      "applies": "d-tooltip",
+      "description": "Hides the tooltip."
+    },
+    {
+      "class": "d-tooltip--hover",
+      "applies": "d-btn",
+      "description": "Applies the tooltip's visibility on :hover to its trigger. This is a CSS-only solution and not recommended."
```
Comment (ryandialpad) [inline docs/_data/tooltips.json]: I would also make a note that this class is applied to the element which you want to trigger to tooltip on hover. I guess applies would signify that if I am assuming correctly?

File: docs/_data/tooltips.json [hunk 2]
```diff
@@ -0,0 +1,46 @@
+{
+  "classes": [
+    {
+      "class": "d-tooltip",
+      "applies": "N/A",
+      "description": "Parent wrapper class for tooltips."
```
Comment (ryandialpad) [inline docs/_data/tooltips.json]: Maybe mention that this class contains the base styles for the tooltip component. We want to make sure that users don't confuse the use of this class and wrap the anchor in it as well

File: docs/_data/tooltips.json [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+    },
+    {
+      "class": "d-tooltip--hover",
+      "applies": "d-btn",
+      "description": "Applies the tooltip's visibility on :hover to its trigger. This is a CSS-only solution and not recommended."
+    },
+    {
+      "class": "d-tooltip--inverted",
+      "applies": "d-tooltip",
+      "description": "Inverts all child styles to work on dark backgrounds."
```
Comment (ryandialpad) [inline docs/_data/tooltips.json]: This class actually modifies the d-tooltip class by inverting the colours, the child classes are responsible for the arrows, position and visibility

**Comment by ryandialpad** [comment]
> @ryandialpad what I'd particularly like your blessing on is whether I accurately listed and described the necessary CSS classes in the "Classes" table. Obviously any other thoughts or feedback are welcome if you have 'em.

A few comments and questions regarding the descriptions. Other than those the docs / descriptions look good :)

I wasn't able to review any of the other changes as it appears that there is a mixture of changes from the branch that I had merged into `v-component-tooltip` and potentially changes from this branch. If possible, you might want to make sure this branch is up to date with the branch that it is going to be merged into so that you can see a full list of changes that will be applied.

I am also assuming that we are going to merge the `v6-component-tooltip` branch into version 6 once this is complete, otherwise I would suggest leaving changes to the breadcrumbs and banners docs to a different branch / PR as to not increase the scope of the PR / branch

**Comment by dialpad-drew** [comment]
> > @ryandialpad what I'd particularly like your blessing on is whether I accurately listed and described the necessary CSS classes in the "Classes" table. Obviously any other thoughts or feedback are welcome if you have 'em.
> 
> A few comments and questions regarding the descriptions. Other than those the docs / descriptions look good :)
> 
> I wasn't able to review any of the other changes as it appears that there is a mixture of changes from the branch that I had merged into `v-component-tooltip` and potentially changes from this branch. If possible, you might want to make sure this branch is up to date with the branch that it is going to be merged into so that you can see a full list of changes that will be applied.
> 
> I am also assuming that we are going to merge the `v6-component-tooltip` branch into version 6 once this is complete, otherwise I would suggest leaving changes to the breadcrumbs and banners docs to a different branch / PR as to not increase the scope of the PR / branch

I branched off `v6-component-tooltip` and then backmerged `version6` because it had recently changes necessary for the documentation changes I made. In the future, how would you recommend I get recent changes from a destination branch to a working branch if not this way?

re: breadcrumbs and banners, that's a good call. I didn't realize I would be sharing this project when I did those changes a couple weeks ago and was cutting corners. I'll clean up my act for next time. 😄

**Comment by ryandialpad** [comment]
> > > @ryandialpad what I'd particularly like your blessing on is whether I accurately listed and described the necessary CSS classes in the "Classes" table. Obviously any other thoughts or feedback are welcome if you have 'em.
> > 
> > 
> > A few comments and questions regarding the descriptions. Other than those the docs / descriptions look good :)
> > I wasn't able to review any of the other changes as it appears that there is a mixture of changes from the branch that I had merged into `v-component-tooltip` and potentially changes from this branch. If possible, you might want to make sure this branch is up to date with the branch that it is going to be merged into so that you can see a full list of changes that will be applied.
> > I am also assuming that we are going to merge the `v6-component-tooltip` branch into version 6 once this is complete, otherwise I would suggest leaving changes to the breadcrumbs and banners docs to a different branch / PR as to not increase the scope of the PR / branch
> 
> I branched off `v6-component-tooltip` and then backmerged `version6` because it had recently changes necessary for the documentation changes I made. In the future, how would you recommend I get recent changes from a destination branch to a working branch if not this way?
> 
> re: breadcrumbs and banners, that's a good call. I didn't realize I would be sharing this project when I did those changes a couple weeks ago and was cutting corners. I'll clean up my act for next time. 😄

No worries :)

Re: Branching. What I usually do is branch off of the feature branch (which is up to date with the branch that it is going to be merged into) and then I merge the feature branch back into my branch just before creating a PR. That way it should only show changes that are being applied to the feature branch. The risk is that if somehow the two have become out of sync, you will now have merge conflicts or unintended changes. To address the risks, I always look at the diffs locally before commit changes.

Target Branch -> Feature Branch -> Some Implementation Branch

**Comment by dialpad-drew** [comment]
> > > > @ryandialpad what I'd particularly like your blessing on is whether I accurately listed and described the necessary CSS classes in the "Classes" table. Obviously any other thoughts or feedback are welcome if you have 'em.
> > > 
> > > 
> > > A few comments and questions regarding the descriptions. Other than those the docs / descriptions look good :)
> > > I wasn't able to review any of the other changes as it appears that there is a mixture of changes from the branch that I had merged into `v-component-tooltip` and potentially changes from this branch. If possible, you might want to make sure this branch is up to date with the branch that it is going to be merged into so that you can see a full list of changes that will be applied.
> > > I am also assuming that we are going to merge the `v6-component-tooltip` branch into version 6 once this is complete, otherwise I would suggest leaving changes to the breadcrumbs and banners docs to a different branch / PR as to not increase the scope of the PR / branch
> > 
> > 
> > I branched off `v6-component-tooltip` and then backmerged `version6` because it had recently changes necessary for the documentation changes I made. In the future, how would you recommend I get recent changes from a destination branch to a working branch if not this way?
> > re: breadcrumbs and banners, that's a good call. I didn't realize I would be sharing this project when I did those changes a couple weeks ago and was cutting corners. I'll clean up my act for next time. 😄
> 
> No worries :)
> 
> Re: Branching. What I usually do is branch off of the feature branch (which is up to date with the branch that it is going to be merged into) and then I merge the feature branch back into my branch just before creating a PR. That way it should only show changes that are being applied to the feature branch. The risk is that if somehow the two have become out of sync, you will now have merge conflicts or unintended changes. To address the risks, I always look at the diffs locally before commit changes.
> 
> Target Branch -> Feature Branch -> Some Implementation Branch

So, in this example in which I have changes on the target branch that I need to use in the implementation branch, I should have gotten those changes into the feature branch and the implementation branch, rather than just the implementation branch?

**Comment by ryandialpad** [comment]
> > > > > @ryandialpad what I'd particularly like your blessing on is whether I accurately listed and described the necessary CSS classes in the "Classes" table. Obviously any other thoughts or feedback are welcome if you have 'em.
> > > > 
> > > > 
> > > > A few comments and questions regarding the descriptions. Other than those the docs / descriptions look good :)
> > > > I wasn't able to review any of the other changes as it appears that there is a mixture of changes from the branch that I had merged into `v-component-tooltip` and potentially changes from this branch. If possible, you might want to make sure this branch is up to date with the branch that it is going to be merged into so that you can see a full list of changes that will be applied.
> > > > I am also assuming that we are going to merge the `v6-component-tooltip` branch into version 6 once this is complete, otherwise I would suggest leaving changes to the breadcrumbs and banners docs to a different branch / PR as to not increase the scope of the PR / branch
> > > 
> > > 
> > > I branched off `v6-component-tooltip` and then backmerged `version6` because it had recently changes necessary for the documentation changes I made. In the future, how would you recommend I get recent changes from a destination branch to a working branch if not this way?
> > > re: breadcrumbs and banners, that's a good call. I didn't realize I would be sharing this project when I did those changes a couple weeks ago and was cutting corners. I'll clean up my act for next time. 😄
> > 
> > 
> > No worries :)
> > Re: Branching. What I usually do is branch off of the feature branch (which is up to date with the branch that it is going to be merged into) and then I merge the feature branch back into my branch just before creating a PR. That way it should only show changes that are being applied to the feature branch. The risk is that if somehow the two have become out of sync, you will now have merge conflicts or unintended changes. To address the risks, I always look at the diffs locally before commit changes.
> > Target Branch -> Feature Branch -> Some Implementation Branch
> 
> So, in this example in which I have changes on the target branch that I need to use in the implementation branch, I should have gotten those changes into the feature branch and the implementation branch, rather than just the implementation branch?

Correct :) We bubble up changes through the feature branch into our implementation branch to prevent the risk of the branches becoming out of sync

So you would have merged `version6` into `v6-component-tooltip` and then merged `v6-component-tooltip` into this branch. And in theory, the diff should only show changes vs the feature branch

**Comment by dialpad-drew** [comment]
> Correct :) We bubble up changes through the feature branch into our implementation branch to prevent the risk of the branches becoming out of sync
> 
> So you would have merged `version6` into `v6-component-tooltip` and then merged `v6-component-tooltip` into this branch. And in theory, the diff should only show changes vs the feature branch

Ahh, makes sense. Thanks for walking through that!

**Review by ryandialpad — APPROVED** [review]
👍 

Great work!

**Pull request merged**